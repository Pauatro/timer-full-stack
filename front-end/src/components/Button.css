button{
    background-color: var(--button-primary-color);
    padding: 0.75rem 3rem 0.75rem 3rem;
    font-size: 1rem;
    margin: 0;
    border: none;
    border-radius: 0.5rem;
    color: var(--font-secondary-color)
}

.timer-button-content{
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
}