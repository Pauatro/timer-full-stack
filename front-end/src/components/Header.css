.header-container{
    height: 8vh;
    width: 100vw;
    padding: 2vh;
    box-sizing: inherit;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
}

.theme-selection-container{
    height: 4vh;
    width: 16vh;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
}