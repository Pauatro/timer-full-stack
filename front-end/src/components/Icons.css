.header-logo, .sun-icon, .moon-icon{
    height: 4vh;
}

.play-icon, .stop-icon{
    height: 1rem;
    margin-right: 1.5rem;
}