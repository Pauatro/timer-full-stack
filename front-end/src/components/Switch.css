.toggle-switch-container{
    height: 100%;
    flex-grow: 1;
    margin-right: 2.5%;
    margin-left: 2.5%;
}

.toggle-switch {
    position: relative;
    height: 100%;
    width: 100%;
}

.toggle-switch-checkbox {
    display: none;
}

.toggle-switch-label {
    display: block;
    overflow: hidden;
    cursor: pointer;
    border-radius: 2vh;
    height: 100%;
    width: 100%; 
}

.toggle-switch-inner {
    display: block;
    width: 100%;
    height: 100%;
    background-color: var(--switch-primary-color);
    overflow: hidden;
}

.toggle-switch-inner:before,
.toggle-switch-inner:after {
  display: block;
  float: left;
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  background-color: var(--switch-secondary-color);
  color: var(--switch-secondary-color);
  content: "N";
}

.toggle-switch-switch {
    display: block;
    position: absolute;
    width: 42.5%;
    height: 80%;
    margin: 5%;
    background: var(--switch-primary-color);
    top: 0;
    bottom: 0;
    right: 47.5%;
    border-radius: 2vh;
    transition: all 0.3s ease-in 0s;
}

.toggle-switch-checkbox:checked  + .toggle-switch-label > .toggle-switch-switch{
    right: 0px;
}