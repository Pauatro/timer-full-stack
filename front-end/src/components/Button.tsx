import React from 'react'
import { EThemes } from '../modules/theme/theme.interfaces'
import { transformMsToTimeString } from '../utils/formatting'
import './Button.css'
import { PlayIcon } from './Icons'

interface IButtonProps{
    onClick: ()=>void;
    children?: JSX.Element;
}

const Button = (props: IButtonProps)=>{
    return <button onClick = {props.onClick}>{props.children}</button>
}

interface ITimerButtonProps{
    time: number;
    theme: EThemes;
}

const TimerButton = (props: ITimerButtonProps)=>{

    return (
        <Button onClick = {()=>{}}>
            <div className = "timer-button-content">
                <PlayIcon theme = {props.theme}/>
                <div>
                    {transformMsToTimeString(props.time)}
                </div>
            </div>
        </Button>
    )
}

export {
    Button,
    TimerButton
}