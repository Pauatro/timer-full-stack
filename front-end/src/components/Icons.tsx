import React from 'react';
import './Icons.css';
import lightLogo from '../resources/logo-light.svg'
import darkLogo from '../resources/logo-dark.svg'
import lightMoonIcon from '../resources/moon-light.svg'
import darkMoonIcon from '../resources/moon-dark.svg'
import lightSunIcon from '../resources/sun-light.svg'
import darkSunIcon from '../resources/sun-dark.svg'
import { EThemes } from '../modules/theme/theme.interfaces'

interface IconProps{
    class: string;
    icon: string;
    alt: string;
}

const Icon = (props: IconProps)=>{
    return (
        <img 
            className = {props.class} 
            src = {props.icon} 
            alt = {props.alt}
        />
    )
}

interface ThemedIconProps{
    theme: EThemes;
}

const HeaderLogos = {
    [EThemes.DARK]: darkLogo,
    [EThemes.LIGHT]: lightLogo
}

const HeaderLogo = (props: ThemedIconProps)=>{
    return (
        <Icon 
            class = "header-logo" 
            icon = {HeaderLogos[props.theme]??lightLogo} 
            alt = "MVST"
        />
    )
}

const SunIcons = {
    [EThemes.DARK]: darkSunIcon,
    [EThemes.LIGHT]: lightSunIcon
}

const SunIcon = (props: ThemedIconProps)=>{
    return (
        <Icon 
            class = "sun-icon" 
            icon = {SunIcons[props.theme]??lightLogo} 
            alt = "Sun"
        />
    )
}

const MoonIcons = {
    [EThemes.DARK]: darkMoonIcon,
    [EThemes.LIGHT]: lightMoonIcon
}

const MoonIcon = (props: ThemedIconProps)=>{
    return (
        <Icon 
            class = "moon-icon" 
            icon = {MoonIcons[props.theme]??lightLogo} 
            alt = "MVST"
        />
    )
}

export {
    HeaderLogo,
    SunIcon,
    MoonIcon
}