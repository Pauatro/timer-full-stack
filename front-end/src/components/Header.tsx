import React from 'react';
import './Header.css';
import { EThemes } from '../modules/theme/theme.interfaces';
import { HeaderLogo, MoonIcon, SunIcon } from './Icons';
import Switch from './Switch';

const Header = () => {
    return  <header className="header-container">
        <HeaderLogo theme = {EThemes.LIGHT}/>
        <div className = "theme-selection-container">
            <MoonIcon theme = {EThemes.LIGHT}/>
            <Switch name = "header-switch"/>
            <SunIcon theme = {EThemes.LIGHT}/>
        </div>
    </header>
}

export default Header