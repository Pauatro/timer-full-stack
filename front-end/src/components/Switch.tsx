import React, { useContext, useState } from 'react';
import { ThemeContext } from '../modules/theme/theme.context';
import './Switch.css'

const Switch = () => {

    const { switchTheme } = useContext(ThemeContext)
    const [ checked, setChecked ] = useState(true)

    const switchChecked = ()=>{
        setChecked(!checked)
        switchTheme()
    }

    return (
        <div className="toggle-switch-container">
            <div className="toggle-switch">
                <input
                    onChange = {switchChecked}
                    type="checkbox"
                    className="toggle-switch-checkbox"
                    name={"theme-switch"}
                    id={"theme-switch"}
                    checked = {checked}
                />
                <label className="toggle-switch-label" htmlFor={"theme-switch"}>
                    <span className="toggle-switch-inner" />
                    <span className="toggle-switch-switch" />
                </label>
            </div>
        </div>
    )
}

export default Switch