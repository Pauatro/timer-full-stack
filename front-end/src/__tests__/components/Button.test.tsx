import React from 'react';
import renderer from 'react-test-renderer';
import { Button, TimerButton } from '../../components/Button';
import { EThemes } from '../../modules/theme/theme.interfaces';

it('Button renders correctly', () => {
  const tree = renderer
    .create(<Button onClick  = {()=>{}}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('TimerButton renders correctly in LIGHT theme', () => {
    const tree = renderer
      .create(<TimerButton 
        time  = {0}
        theme = {EThemes.LIGHT}
        onTimerClick = {()=>{}}
        timerIsActive = {true}
        />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('TimerButton renders correctly in DARK theme', () => {
  const tree = renderer
    .create(<TimerButton 
      time  = {0}
      theme = {EThemes.DARK}
      onTimerClick = {()=>{}}
      timerIsActive = {true}
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});