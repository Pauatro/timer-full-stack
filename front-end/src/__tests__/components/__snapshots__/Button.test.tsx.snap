// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  onClick={[Function]}
/>
`;

exports[`TimerButton renders correctly in DARK theme 1`] = `
<button
  onClick={[Function]}
>
  <div
    className="timer-button-content"
  >
    <img
      alt="Stop"
      className="stop-icon"
      src="stop-icon-dark.svg"
    />
    <div>
      0:00:00
    </div>
  </div>
</button>
`;

exports[`TimerButton renders correctly in LIGHT theme 1`] = `
<button
  onClick={[Function]}
>
  <div
    className="timer-button-content"
  >
    <img
      alt="Stop"
      className="stop-icon"
      src="stop-icon-light.svg"
    />
    <div>
      0:00:00
    </div>
  </div>
</button>
`;
